packages/material-ui-based-components/src/components/CheckboxGroup.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback } from 'react';
+import { withRuleEngine } from '../shared/withRuleEngine';
+import { PROPS } from '../utils/types';
+import { Checkbox, FormControl, FormControlLabel, FormGroup, FormHelperText, FormLabel } from '@mui/material';
+
+const CheckboxGroupComponent = (props: PROPS) => {
+  const {
+    label, id, required, enum: enums, enabled, errorMessage,
+    description, enumNames, isError, properties
+  } = props;
+  const k = 'afs:layout';
+  let row = false;
+  if (properties && properties[k]) {
+    row = properties[k].orientation === 'horizontal';
+  }
+
+  const changeHandler = useCallback((event: any) => {
+    props.dispatchChange(event.target.value);
```

Comment:
please check this method https://github.com/adobe/aem-forms-headless-components/blob/main/packages/react-native/src/components/CheckboxGroup.tsx#L15

Reply:
Checked the method. Improved as required.